### PR TITLE
eVic 662 prepare api for topics

### PR DESCRIPTION
**Pull request by tkuzynow**

### CODE REVIEW

File: src/test/java/de/caritas/cob/consultingtypeservice/api/repository/TopicRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@TestPropertySource(properties = "spring.profiles.active=testing")
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.ANY)
+@ExtendWith(SpringExtension.class)
+@DataJpaTest
+@Disabled
```
Comment (tkuzynow) [inline src/test/java/de/caritas/cob/consultingtypeservice/api/repository/TopicRepositoryTest.java]: we have a separate task to enable it

File: api/topicservice.yml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                $ref: '#/components/schemas/TopicDTO'
+        401:
+          description: UNAUTHORIZED - no/invalid Keycloak token
+        409:
+          description: CONFLICT - unique constraint validation fails
+        500:
+          description: INTERNAL SERVER ERROR - server encountered unexpected condition
+components:
+  schemas:
+    TopicDTO:
```
Comment (maeventer) [inline api/topicservice.yml:119]: I could not exactly remember what we said for hal structure adaptions for topics but i thought we will include hal here i just can´t remember when :-P

File: pom.xml [hunk 3]
```diff
@@ -194,6 +217,38 @@
...
+    </dependency>
+
+    <!-- Test scope dependencies -->
+    <dependency>
+      <groupId>com.h2database</groupId>
+      <artifactId>h2</artifactId>
+      <scope>test</scope>
+    </dependency>
+
+
```
Comment (maeventer) [inline pom.xml]: PLease remove unneccessary line breaks

**Review by maeventer — APPROVED** [review]

File: api/topicservice.yml (hunk 2 above)
Comment (tkuzynow) [reply api/topicservice.yml:119]: we have a separate story for HAL in the backlog, but on the recent planning it was decided not to take it to the sprint

File: pom.xml (hunk 3 above)
Comment (tkuzynow) [reply pom.xml]: done

**Pull request merged**